zson/formatter.go

```diff
@@ -274,27 +274,54 @@ func (f *Formatter) formatRecord(indent int, typ *zed.TypeRecord, bytes zcode.By
 
 func (f *Formatter) formatVector(indent int, open, close string, inner zed.Type, zv zed.Value, known, parentImplied bool) (bool, error) {
 	f.build(open)
-	len, err := zv.ContainerLength()
+	n, err := zv.ContainerLength()
 	if err != nil {
 		return true, err
 	}
-	if len == 0 {
+	if n == 0 {
 		f.build(close)
 		return true, nil
 	}
 	indent += f.tab
 	sep := f.newline
 	it := zv.Iter()
+	union, _ := zed.TypeUnder(inner).(*zed.TypeUnion)
```

Comment:
This will lose the named type.  I think you want to just not do the TypeUnder...

Reply:
You are right that we're losing the name on named unions. Having said that, I don't think we want the individual elements to get decorated as they would if we removed this on a named union here. Maybe we should decorate the entire array if its inner type is a named union?

```
echo '[1(named=(int64,string)),"foo"(named=(int64,string))]' | go run ./cmd/zq -
```
=>
```
[1,"foo"]([=named])
```
?

Reply:
(I added a change so the above example now works.